sklearn/ensemble/_hist_gradient_boosting/histogram.pyx

```diff
@@ -142,11 +146,23 @@ cdef class HistogramBuilder:
             )
             bint has_interaction_cst = allowed_features is not None
             int n_threads = self.n_threads
+            unsigned int n_feature_groups = self.n_features // 4
+            double [:] time_vec = np.zeros(shape=(2,), dtype=np.float64)
```

Comment:
Why do we use heap allocation, here? Can't we allocate on the stack and pass their address using pointers?

Reply:
That's more for benchmarking here. I guess in the unlikely case this PR gets merged, we would remove the timing part.